vividus-facade-jira/src/main/java/org/vividus/jira/JiraFacade.java

```diff
@@ -38,49 +51,147 @@
     private static final String ISSUE = "issue/";
     private static final String ISSUE_ENDPOINT = REST_API_ENDPOINT + ISSUE;
 
-    private final JiraClient jiraClient;
+    private final PropertyMappedCollection<JiraConfiguration> jiraConfigurations;
+    private final IHttpClientFactory httpClientFactory;
+    private final HttpClientConfig defaultHttpClientConfig;
 
-    public JiraFacade(JiraClient jiraClient)
+    private final LoadingCache<String, JiraClient> jiraClients = CacheBuilder.newBuilder().build(new CacheLoader<>()
     {
-        this.jiraClient = jiraClient;
+        @Override
+        public JiraClient load(String key) throws Exception
+        {
+            JiraConfiguration configuration = jiraConfigurations.getNullable(key).get();
+            HttpClientConfig httpConfig = Optional.ofNullable(configuration.getHttpConfig())
```

Comment:
is it allowed to declare `httpConfig` as `Optional<HttpClientConfig>` in `JiraConfiguration` ?